File: 2144-maximum-difference-between-increasing-elements/2144-maximum-difference-between-increasing-elements.c
int maximumDifference(int* nums, int numsSize) {

    int min = nums[0]; //min = i,max = j
    int max = -1;
    for(int i = 1;i<numsSize;i++){
        if(nums[i]>min){
            if(nums[i]-min>max){
                max = nums[i]-min;
            }
        }
        else{
            min = nums[i];
        }
    }
    return max;
}

File: 2144-maximum-difference-between-increasing-elements/2144-maximum-difference-between-increasing-elements.java
class Solution {
    public int maximumDifference(int[] nums) {
        int min=nums[0];
        int maxv=-1;
        for(int i=1;i<nums.length;i++){
            if(nums[i] >min){
                maxv =Math.max(maxv,nums[i]-min);
            }
            else{
                min=nums[i];
            }
        }
        return maxv;
    }
}